posthog/warehouse/data_load/test/test_service.py

```diff
@@ -1,69 +1,197 @@
-        expected = ScheduleCalendarSpec(
-            hour=[ScheduleRange(start=2, end=22, step=5)], minute=[ScheduleRange(start=15, end=15, step=1)]
-        )
-        self.assertEqual(result.hour, expected.hour)
-        self.assertEqual(result.minute, expected.minute)
-
-    def test_late_start_time(self):
-        # Test with 8-hour frequency starting at 22:00, expect 1 execution per day
-        result = get_calendar_spec(22, 0, timedelta(hours=8))
-        expected = ScheduleCalendarSpec(
-            hour=[ScheduleRange(start=22, end=22, step=8)], minute=[ScheduleRange(start=0, end=0, step=1)]
-        )
-        self.assertEqual(result.hour, expected.hour)
-        self.assertEqual(result.minute, expected.minute)
-
-    def test_frequency_larger_than_day(self):
-        result = get_calendar_spec(9, 45, timedelta(hours=36))
-        expected = ScheduleCalendarSpec(
-            hour=[ScheduleRange(start=9, end=9, step=36)], minute=[ScheduleRange(start=45, end=45, step=1)]
-        )
-        self.assertEqual(result.hour, expected.hour)
-        self.assertEqual(result.minute, expected.minute)
+import datetime as dt
+import logging
+import random
+import uuid
+
+import pytest
+import pytest_asyncio
+from asgiref.sync import async_to_sync, sync_to_async
+from temporalio.client import Client as TemporalClient
+from temporalio.service import RPCError
+
+from posthog.models import Organization, Team
+from posthog.temporal.common.client import sync_connect
+from posthog.warehouse.data_load.service import get_sync_schedule
+from posthog.warehouse.models import ExternalDataSchema, ExternalDataSource
+
+pytestmark = [
+    pytest.mark.django_db,
+]
+
+
+@async_to_sync
+async def delete_temporal_schedule(temporal: TemporalClient, schedule_id: str):
+    """Delete a Temporal Schedule with the given id."""
+    handle = temporal.get_schedule_handle(schedule_id)
+    await handle.delete()
+
+
+def cleanup_temporal_schedules(client):
+    """Clean up any Temporal Schedules created during the test."""
+    for schedule in ExternalDataSchema.objects.all():
+        try:
+            delete_temporal_schedule(client, str(schedule.id))
+        except RPCError:
+            # Assume this is fine as we are tearing down, but don't fail silently.
+            logging.warning("Schedule %s has already been deleted, ignoring.", schedule.id)
+            continue
+
+
+@pytest.fixture
+def temporal():
+    """Return a TemporalClient instance."""
+    client = sync_connect()
+    yield client
+    cleanup_temporal_schedules(client)
+
+
+@pytest_asyncio.fixture
+async def organization():
+    """A test organization."""
+    name = f"TestOrg-{random.randint(1, 99999)}"
+    org = await Organization.objects.acreate(name=name, is_ai_data_processing_approved=True, slug=name)
+
+    yield org
+    await org.adelete()
+
+
+@pytest_asyncio.fixture
+async def team(organization):
+    name = f"BatchExportsTestTeam-{random.randint(1, 99999)}"
```

Comment:
`BatchExports` 😉

Reply:
lol, you can see where I took these tests from 🙈